packages/adapter-session-unstorage/src/index.ts

```diff
@@ -0,0 +1,90 @@
+import type {
+	SessionSchema,
+	SessionAdapter,
+	AdapterFunction
+} from "lucia-auth";
+import { type Storage } from "unstorage";
+
+const adapter =
+	(unstorage: {
+		session: Storage;
+		sessionKeyPrefix?: string;
+		userSessionKeyPrefix?: string;
+	}): AdapterFunction<SessionAdapter> =>
+	() => {
+		const {
+			session: sessionUnstorage,
+			sessionKeyPrefix = "session",
+			userSessionKeyPrefix = "user_session"
+		} = unstorage;
+		return {
+			sessionKeyPrefix,
+			userSessionKeyPrefix,
+			getSession: async (sessionId) => {
+				const sessionData = await sessionUnstorage.getItem(
+					`${sessionKeyPrefix}:${sessionId}`
+				);
+				return sessionData as SessionSchema | null;
+			},
+			getSessionsByUserId: async (userId) => {
+				const userKeys = await sessionUnstorage.getKeys(
+					`${userSessionKeyPrefix}:${userId}`
+				);
+				const sessionIds = await Promise.all(
+					userKeys.map((id) => sessionUnstorage.getItem(id) as Promise<string>)
+				);
+				return Promise.all(
+					sessionIds.map(
+						(id) =>
+							sessionUnstorage.getItem(
+								`${sessionKeyPrefix}:${id}`
+							) as Promise<SessionSchema>
+					)
+				);
+			},
+			setSession: async (session) => {
+				await Promise.all([
+					sessionUnstorage.setItem(
+						`${userSessionKeyPrefix}:${session.user_id}:${Date.now()}`,
+						session.id
```

Comment:
How does it handle one-to-many relations (a user can have multiple sessions)?

Reply:
Yes, a user can have multiple sessions (the unix timestamp enforce key uniqueness). This behave in the same way as the Redis adapter.

Reply:
Why the date btw?

Reply:
Fyi, the redis adapter is pretty old and needs a refresh, so I recommend not using it as a reference.